Assets/Scripts/Player.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class Player : MonoBehaviour
+{
+    [SerializeField]
+    private float speed;
+
+    [SerializeField]
+    private float dashSpeed;
+
+    [SerializeField]
+    private float dashDuration;
+
+    [SerializeField]
+    private float dashCooldown;
+
+    [SerializeField]
+    private Rigidbody demoProjectile; // prefab
+
+    private Animator animator;
+    private Rigidbody body;
+    private Transform hand;
+
+    private bool dashOnCooldown;
+    private bool isWalking;
+
+    private NavMeshAgent agent;
+
+    private Vector3 aimDirection;
+
+    private void Awake()
+    {
+        animator = GetComponentInChildren<Animator>();
+        body = GetComponentInChildren<Rigidbody>();
+        hand = transform.Find("Hand");
+        agent = GetComponent<NavMeshAgent>();
+    }
+
+    //player move function
+    public void Move(float horizontal, float vertical)
+    {
+        isWalking = (horizontal != 0 || vertical != 0);
+        animator.SetBool("IsWalking", (horizontal != 0 || vertical != 0));
```

Comment:
Unused variable isWalking. Should probably just remove it

Reply:
whoops. I removed it, then put it back, forgot to add it back to the SetBool function.. might be useful later.